pkgs/applications/misc/keymapp/default.nix

```diff
@@ -0,0 +1,62 @@
+{ stdenv
+, lib
+, fetchurl
+, autoPatchelfHook
+, libusb1
+, webkitgtk
+, gtk3
+, writeShellScript
+}:
+
+stdenv.mkDerivation rec {
+  pname = "keymapp";
+  version = "1.0.5";
+
+  src = fetchurl {
+    url = "https://oryx.nyc3.cdn.digitaloceanspaces.com/keymapp/keymapp-${version}.tar.gz";
+    hash = "sha256-e9Ty3gMb+nkXGK8sND4ljyrIxP+1fLasiV6DoTiWmsU=";
+  };
+
+  nativeBuildInputs = [
+    autoPatchelfHook
```

Comment:
I dont have a ZSA keyboard with me, but on my system, i needed to add `wrapGAppsHook` here, to be able to select files via file dialog (in case one wants to download the firmware and select it via keymapp), i can report back tomorrow, if thats still needed with your package here (some minor stuff differs to what i have in mine)

Reply:
I have not tested this, I just tested it detects keyboard and is able to show layers/layout and keypresses (what I care about).

I flash from chrome, but I will try flashing from keymapp and apply your change if needed.

I'll be able to test it in ~12 hours.

Reply:
so i tested it at home with my moonlander connected, the keymapp crashes as soon as one clicks on select file with 
```
[nix-shell:~/.cache/nixpkgs-review/pr-268299]$ keymapp
Overriding existing handler for signal 10. Set JSC_SIGNAL_FOR_GC if you want WebKit to use a different signal
Could not determine the accessibility bus address

(keymapp_bin:16664): GLib-GIO-ERROR **: 20:11:39.355: No GSettings schemas are installed on the system
SIGTRAP: trace trap
PC=0x7fe558180b87 m=0 sigcode=128
signal arrived during cgo execution

goroutine 1 [syscall, locked to thread]:
runtime.cgocall(0x822130, 0xc0001edb88)
        /usr/local/go/src/runtime/cgocall.go:157 +0x4b fp=0xc0001edb60 sp=0xc0001edb28 pc=0x41a96b
github.com/wailsapp/wails/v2/internal/frontend/desktop/linux._Cfunc_Opendialog(0xc0007d0900)
        _cgo_gotypes.go:643 +0x3f fp=0xc0001edb88 sp=0xc0001edb60 pc=0x80309f
github.com/wailsapp/wails/v2/internal/frontend/desktop/linux.(*Window).OpenFileDialog.func4.1(0x1?)
        /go/pkg/mod/github.com/wailsapp/wails/v2@v2.6.0/internal/frontend/desktop/linux/window.go:405 +0x3e fp=0xc0001edbc0 sp=0xc0001edb88 pc=0x8104be
github.com/wailsapp/wails/v2/internal/frontend/desktop/linux.(*Window).OpenFileDialog.func4()
        /go/pkg/mod/github.com/wailsapp/wails/v2@v2.6.0/internal/frontend/desktop/linux/window.go:405 +0x17 fp=0xc0001edbd8 sp=0xc0001edbc0 pc=0x810457

```
The log prior the stack shows that its complaining about the GSettings schema missing. other apps are also known to crash by that error.
On the package in my personal nix config, that went away by adding gapps hook, possibly here some extra work is needed, when having the wrapper script inbetween, so we should prevent double wrapping here.

Reply:
selecting the firmware now works fine for me with the `wrapGAppsHook` change

Reply:
So is the double-wrap still a problem or not?
If this works I'd just ship it as is.

Reply:
I switched to using proper wrapping using nix utils. So it's simpler and possibly that wrapper should work better.

I added an icon as you suggested and it works fine with it.

I also made sure I set the env variable only as a default so users can easily override it if needed.

Reply:
Ah sorry, It't late here. I completely missed your other suggestion https://github.com/NixOS/nixpkgs/pull/268299#discussion_r1409768020

I'll do it like that I didn't realize I can tweak GAppsWrapper so easily.